test/e2e/common/networking.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	. "github.com/onsi/ginkgo"
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/kubernetes/test/e2e/framework"
+	networking_util "k8s.io/kubernetes/test/utils"
+)
+
+var _ = framework.KubeDescribe("Networking", func() {
+	f := framework.NewDefaultFramework("pod-network-test")
+	BeforeEach(func() {
+	})
+
+	framework.KubeDescribe("Granular Checks: Pods", func() {
+
+		// Try to hit all endpoints through a test container, retry 5 times,
+		// expect exactly one unique hostname. Each of these endpoints reports
+		// its own hostname.
+		It("should function for intra-pod communication: http [Conformance]", func() {
+			config := networking_util.NewNetworkingTestNodeE2EConfig(f)
```

Comment:
The name is a little confusing, because common test also runs in cluster e2e. Maybe `NewCoreNetworkingTestConfig` or something?